From b6db5ab4f5a2c0de385768059bb007ec650e3e42 Mon Sep 17 00:00:00 2001
From: WUKAWL <21525471@qq.com>
Date: Thu, 3 Apr 2025 22:00:26 +0800
Subject: [PATCH 15/25] Fix touch on MEIZU 21 Pro (#1)

* Fix touch on Meizu 21 Pro

Raw values from the panel are multiplied by 10,
causing most touch events to fall out of bounds and get dropped

* Fix touch on Meizu 21 Pro

Raw values from the panel are multiplied by 10, causing most touch events to fall out of bounds and get dropped
---
 .../reader/mapper/accumulator/MultiTouchMotionAccumulator.cpp   | 2 +-
 .../reader/mapper/accumulator/SingleTouchMotionAccumulator.cpp  | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/services/inputflinger/reader/mapper/accumulator/MultiTouchMotionAccumulator.cpp b/services/inputflinger/reader/mapper/accumulator/MultiTouchMotionAccumulator.cpp
index 8710abe..2e7f6b4 100644
--- a/services/inputflinger/reader/mapper/accumulator/MultiTouchMotionAccumulator.cpp
+++ b/services/inputflinger/reader/mapper/accumulator/MultiTouchMotionAccumulator.cpp
@@ -159,7 +159,7 @@ void MultiTouchMotionAccumulator::populateCurrentSlot(
 
 MultiTouchMotionAccumulator::Slot::Slot() {
     std::string targetDevice = android::base::GetProperty("ro.product.vendor.device", "");
-    if (targetDevice == "meizu21" || targetDevice == "Meizu21Note") {
+    if (targetDevice == "meizu21" || targetDevice == "meizu21Pro" || targetDevice == "Meizu21Note") {
         mAbsMtPositionXYRatio = 10;
     }
 }
diff --git a/services/inputflinger/reader/mapper/accumulator/SingleTouchMotionAccumulator.cpp b/services/inputflinger/reader/mapper/accumulator/SingleTouchMotionAccumulator.cpp
index 72d8cd3..4c91b19 100644
--- a/services/inputflinger/reader/mapper/accumulator/SingleTouchMotionAccumulator.cpp
+++ b/services/inputflinger/reader/mapper/accumulator/SingleTouchMotionAccumulator.cpp
@@ -27,7 +27,7 @@ SingleTouchMotionAccumulator::SingleTouchMotionAccumulator() {
     clearAbsoluteAxes();
 
     std::string targetDevice = android::base::GetProperty("ro.product.vendor.device", "");
-    if (targetDevice == "meizu21" || targetDevice == "Meizu21Note") {
+    if (targetDevice == "meizu21" || targetDevice == "meizu21Pro" || targetDevice == "Meizu21Note") {
         mAbsXYRatio = 10;
     }
 }
-- 
2.34.1

